openedx/features/course_experience/tests/views/test_course_home.py

```diff
@@ -448,12 +448,8 @@ def test_course_does_not_expire_for_verified_user(self):
         url = course_home_url(course)
 
         user = UserFactory.create(password=self.TEST_PASSWORD)
-        ScheduleFactory(
-            start_date=THREE_YEARS_AGO,
-            enrollment__mode=CourseMode.VERIFIED,
-            enrollment__course_id=course.id,
-            enrollment__user=user
-        )
+        CourseEnrollment.enroll(user, self.course.id, mode=CourseMode.VERIFIED)
+        Schedule.objects.update(start_date=THREE_YEARS_AGO)
```

Comment:
Is there any chance this is talking about schedules that aren't the one you just created for this user? Is that a problem? Does `.enroll` return an enrollment that you could then do `enrollment.schedule.start_date` or something?

Reply:
I think... the relational object wasn't defined on the returned enrollment, because the schedule is created via signal. I believe that's why I did this?

Regardless, there aren't other schedules - it's a test so it has a fresh db. I could avoid that concern by adding `.filter(enrollment=enrollment)` to the update line, but it's just test code.